ui/packages/consul-ui/app/templates/dc/peers/index.hbs

```diff
@@ -0,0 +1,59 @@
+<Route @name={{route}} as |route|>
+  <Watcher @watch={{this.model.loadPeers}} as |w|>
+    {{did-insert w.fns.start}}
+    {{will-destroy w.fns.stop}}
+  </Watcher>
```

Comment:
Are disconnection states something you are thinking of adding here? We have an idea of "disconnection" being the state after at least one successful HTTP request, but then erroring on a subsequent one. When this happens we show a notification at the top of the UI and then keep the last set of loaded data on the page so the user can still see the last thing they loaded.

Reply:
👷‍♂️ That's why I was saying this component would ideally use a statechart to model its behavior - additional requirements tend to creep in into components like this and the component already is implementing rather intricate behavior. I don't know enough about the requirements you outlined in your comment so I suggest to add this as a follow up. Sounds like a useful addition though and it definitely sounds like something where requirements should be fleshed out explicitly before jumping into implementation.